core/src/main/java/org/hisp/dhis/android/sdk/synchronization/domain/trackedentityinstance/TrackedEntityInstanceSynchronizer.java

```diff
@@ -1,10 +1,138 @@
 package org.hisp.dhis.android.sdk.synchronization.domain.trackedentityinstance;
 
+import static android.R.attr.id;
+
+import static org.hisp.dhis.android.sdk.persistence.models.FailedItem.TRACKEDENTITYINSTANCE;
+
+import org.hisp.dhis.android.sdk.network.APIException;
+import org.hisp.dhis.android.sdk.network.response.ImportSummary2;
+import org.hisp.dhis.android.sdk.persistence.models.Enrollment;
+import org.hisp.dhis.android.sdk.persistence.models.ImportSummary;
+import org.hisp.dhis.android.sdk.persistence.models.Relationship;
 import org.hisp.dhis.android.sdk.persistence.models.TrackedEntityInstance;
+import org.hisp.dhis.android.sdk.synchronization.domain.common.Synchronizer;
+import org.hisp.dhis.android.sdk.synchronization.domain.enrollment.EnrollmentSynchronizer;
+import org.hisp.dhis.android.sdk.synchronization.domain.enrollment.IEnrollmentRepository;
+import org.hisp.dhis.android.sdk.synchronization.domain.event.IEventRepository;
+import org.hisp.dhis.android.sdk.synchronization.domain.faileditem.IFailedItemRepository;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TrackedEntityInstanceSynchronizer extends Synchronizer{
+
+
+    ITrackedEntityInstanceRepository mTrackedEntityInstanceRepository;
+    IEnrollmentRepository mEnrollmentRepository;
+    IEventRepository mEventRepository;
+    IFailedItemRepository mFailedItemRepository;
 
-public class TrackedEntityInstanceSynchronizer {
+    public TrackedEntityInstanceSynchronizer(ITrackedEntityInstanceRepository trackedEntityInstanceRepository, IEnrollmentRepository enrollmentRepository, IEventRepository eventRepository,
+            IFailedItemRepository failedItemRepository) {
+        super(failedItemRepository);
+        mTrackedEntityInstanceRepository = trackedEntityInstanceRepository;
+        mEnrollmentRepository = enrollmentRepository;
+        mEventRepository = eventRepository;
+        mFailedItemRepository = failedItemRepository;
+    }
 
     public void sync(TrackedEntityInstance trackedEntityInstance) {
-        //mTrackedEntityInstanceRepository.Sync(trackedEntityInstance);
+
+        if(trackedEntityInstance.getRelationships()!=null){
+            Map<String, TrackedEntityInstance> relatedTeiList = new HashMap<>();
+            mTrackedEntityInstanceRepository.getRecursiveRelationatedTeis(trackedEntityInstance, relatedTeiList);
```

Comment:
@idelcano the code to retrieve relational TEIS is complicated and is easy contains errors. I think that we should simplify to if a TEI to sync has relations directly retrieve all local TEIS with fromServer equal a false and sync it in batch mode. It´s simply remove getRecursiveRelationatedTeis method, What do you think about this?